core/src/test/java/org/apache/iceberg/view/TestViewMetadata.java

```diff
@@ -63,8 +62,11 @@ public void testExpiration() {
     ViewVersion v2 = newViewVersion(2, "select count(1) as count from t2");
     Map<Integer, ViewVersion> versionsById = ImmutableMap.of(1, v1, 2, v2, 3, v3);
 
-    List<ViewVersion> retainedVersions = ViewMetadata.Builder.expireVersions(versionsById, 2);
-    assertThat(retainedVersions).hasSameElementsAs(ImmutableList.of(v2, v3));
+    assertThat(ViewMetadata.Builder.expireVersions(versionsById, 3))
+        .containsExactlyInAnyOrder(v1, v2, v3);
```

Comment:
@nastra, why did this change to `containsExactlyInAnyOrder`? Shouldn't the order be predictable?

Reply:
This is what I mentioned in https://github.com/apache/iceberg/pull/9725#discussion_r1489628489. The order here was always reversed (aka `versions` were 3, 2, 1 instead of 1, 2, 3) but we actually never tested this, because `hasSameElementsAs` only checks that the same elements are existing (without checking their order)